guides/source/active_record_basics.md

```diff
@@ -71,88 +67,105 @@ Convention over Configuration in Active Record
+* `(association_name)_type` - Stores the type for [polymorphic
+  associations](association_basics.html#polymorphic-associations).
 * `(table_name)_count` - Used to cache the number of belonging objects on
-  associations. For example, a `comments_count` column in an `Article` class that
-  has many instances of `Comment` will cache the number of existent comments
-  for each article.
+  associations. For example, a `comments_count` column in an `Article` class
+  that has many instances of `Comment` will cache the number of existent
+  comments for each article.
 
-NOTE: While these column names are optional, they are in fact reserved by Active Record. Steer clear of reserved keywords unless you want the extra functionality. For example, `type` is a reserved keyword used to designate a table using Single Table Inheritance (STI). If you are not using STI, try an analogous keyword like "context", that may still accurately describe the data you are modeling.
+NOTE: While these column names are optional, they are reserved by Active Record.
+Steer clear of reserved keywords when naming your table's columns. For example,
+`type` is a reserved keyword used to designate a table using Single Table
+Inheritance (STI). If you are not using STI, use a different word to accurately
+describe the data you are modeling.
 
 Creating Active Record Models
 -----------------------------
 
-When generating an application, an abstract `ApplicationRecord` class will be
-created in `app/models/application_record.rb`. This is the base class for all
-models in an app, and it's what turns a regular ruby class into an Active Record
-model.
+When generating a Rails application, an abstract `ApplicationRecord` class will
+be created in `app/models/application_record.rb`. The `ApplicationRecord` class
+inherits from [`ActiveRecord::Base`](https://api.rubyonrails.org/classes/ActiveRecord/Base.html) and it's what turns a regular ruby class into
+an Active Record model.
 
-To create Active Record models, subclass the `ApplicationRecord` class and you're good to go:
+`ApplicationRecord` is the base class for all models in your app. To create
+Active Record models, subclass the `ApplicationRecord` class and you're good to
+go:
 
 ```ruby
 class Product < ApplicationRecord
 end
 ```
 
-This will create a `Product` model, mapped to a `products` table at the
-database. By doing this you'll also have the ability to map the columns of each
-row in that table with the attributes of the instances of your model. Suppose
-that the `products` table was created using an SQL (or one of its extensions) statement like:
+This will create a `Product` model, mapped to a `products` table in the
+database. Using `ApplicationRecord`, each column in the table is mapped to attributes of the `Product` class. And an instance of the `Product` can represent a row in the `products` table. The `products` table can be created using an SQL statement like this:
```

Comment:
We might be able to skip another mention of `ApplicationRecord`:

```suggestion
This will create a `Product` model, mapped to a `products` table in the
database, where each column in the table is mapped to attributes of the `Product` class. An instance of `Product` can represent a row in the `products` table. The `products` table can be created using an SQL statement like this:
```